dashboard/app/models/foorm/form.rb

```diff
@@ -84,4 +90,59 @@ def self.fill_in_library_items(questions)
     end
     return questions
   end
+
+  def self.get_matrix_question_id(parent_question_id, sub_question_id)
+    parent_question_id + '-' + sub_question_id
+  end
+
+  # For a given Form, this method will produce a CSV of all responses
+  # received for that Form. It includes the content of the form submitted
+  # by a user, as well as some additional metadata about the context
+  # in which the form was submitted (eg, workshop ID, user ID).
+  def submissions_to_csv(file_path)
+    formatted_submissions = []
+
+    submissions.each do |submission|
+      formatted_submission = submission.formatted_answers
+      formatted_submission.merge! submission.formatted_metadata
+
+      formatted_submissions << formatted_submission
+    end
+
+    CSV.open(file_path, "wb") do |csv|
+      break if formatted_submissions.empty?
+
+      headers = readable_questions
+      headers_with_config_variables = formatted_submissions.first.keys.map do |question_id|
```

Comment:
will this only include questions that were answered on the first submission?

Reply:
Fixed, I think.